hedera-node/hedera-app/src/main/java/com/hedera/node/app/tss/handlers/TssSubmissions.java

```diff
@@ -138,6 +156,16 @@ private CompletableFuture<Void> submit(
                 submissionExecutor);
     }
 
+    private Instant nextValidStartFor(@NonNull final HandleContext context) {
+        if (lastContextUsed != context) {
+            lastContextUsed = context;
+            nextOffset.set(0);
+        } else {
+            nextOffset.incrementAndGet();
+        }
+        return context.consensusNow().plusNanos(nextOffset.get());
```

Comment:
If the code is using an executor to submit tss messages, shouldn't `lastContextUsed` be threadsafe?

Reply:
The handler will synchronously schedule all its TSS Message submissions in a single context, so this is just a trivial way to ensure each scheduled submission uses a unique TransactionID.

Reply:
Ok. I remembered this is called by the handling thread anyway. Thanks!